src/libraries/Common/src/System/Security/Cryptography/DSAAndroid.cs

```diff
@@ -0,0 +1,449 @@
+                Span<byte> signDestination = stackalloc byte[SignatureStackBufSize];
+
+#if INTERNAL_ASYMMETRIC_IMPLEMENTATIONS
+                if (signatureFormat == DSASignatureFormat.IeeeP1363FixedFieldConcatenation)
+#endif
+                {
+                    int fieldSizeBytes = Interop.AndroidCrypto.DsaSignatureFieldSize(key);
+                    int p1363SignatureSize = 2 * fieldSizeBytes;
+
+                    if (destination.Length < p1363SignatureSize)
+                    {
+                        bytesWritten = 0;
+                        return false;
+                    }
+
+                    int fieldSizeBits = fieldSizeBytes * 8;
+                    ReadOnlySpan<byte> derSignature = SignHash(hash, signDestination, maxSignatureSize, key);
+                    bytesWritten = AsymmetricAlgorithmHelpers.ConvertDerToIeee1363(derSignature, fieldSizeBits, destination);
+                    Debug.Assert(bytesWritten == p1363SignatureSize);
+                    return true;
+                }
+#if INTERNAL_ASYMMETRIC_IMPLEMENTATIONS
+                else if (signatureFormat == DSASignatureFormat.Rfc3279DerSequence)
+                {
+                    if (destination.Length >= maxSignatureSize)
+                    {
+                        signDestination = destination;
+                    }
+                    else if (maxSignatureSize > signDestination.Length)
+                    {
+                        Debug.Fail($"Stack-based signDestination is insufficient ({maxSignatureSize} needed)");
+                        bytesWritten = 0;
+                        return false;
+                    }
+
+                    ReadOnlySpan<byte> derSignature = SignHash(hash, signDestination, maxSignatureSize, key);
+
+                    if (destination == signDestination)
+                    {
+                        bytesWritten = derSignature.Length;
+                        return true;
+                    }
+
+                    return Helpers.TryCopyToDestination(derSignature, destination, out bytesWritten);
+                }
+                else
+                {
+                    Debug.Fail($"Missing internal implementation handler for signature format {signatureFormat}");
+                    throw new CryptographicException(
+                        SR.Cryptography_UnknownSignatureFormat,
+                        signatureFormat.ToString());
+                }
+#endif
+            }
+    internal static string ByteArrayToHex(byte[] bytes)
+    {
+        return ByteArrayToHex((ReadOnlySpan<byte>)bytes);
+    }
+
+    internal static string ByteArrayToHex(Span<byte> bytes)
+    {
+        return ByteArrayToHex((ReadOnlySpan<byte>)bytes);
+    }
+
+    internal static string ByteArrayToHex(ReadOnlyMemory<byte> bytes)
+    {
+        return ByteArrayToHex(bytes.Span);
+    }
+
+    internal static string ByteArrayToHex(ReadOnlySpan<byte> bytes)
```

Comment:
I don't see these being used any more sense the `Debug.WriteLine`s were removed. Should these go away too?